recipes/octave/meta.yaml

```diff
@@ -0,0 +1,104 @@
+{% set version = "4.2.1" %}
+{% set variant = "openblas" %}
+
+package:
+  name: octave
+  version: {{ version }}
+
+source:
+  url: https://ftp.gnu.org/gnu/octave/octave-{{ version }}.tar.gz
+  sha256: 80c28f6398576b50faca0e602defb9598d6f7308b0903724442c2a35a605333b
+  patches:
+    # patches taken from macports
+    # https://github.com/macports/macports-ports/tree/master/math/octave/files
+    # Link fails with undefined symbol ~file_stat
+    - patch-file-stat.cc.diff
+    # qscintilla2 library is not found without this
+    - patch-qscintilla2.diff
+
+build:
+  number: 0
+  skip: True  # [win or osx]
+  features:
+    - blas_{{ variant }}
+
+requirements:
+  build:
+    - toolchain
+    - gcc          # [osx]
+    - sed
+    - perl 5.20.3.1
+    - openblas 0.2.19|0.2.19.*
+    - blas 1.1 {{ variant }}
+    - pcre 8.37
+    - readline 6.2*
+    - arpack 3.4.*
+    - curl 7.48.*
+    - fftw 3.3.*
+    - fontconfig 2.12.*
+    - freetype 2.7|2.7.*
+    - glpk 4.61
+    - gnuplot 5.0.*
+    - hdf5 1.8.17|1.8.17.*
+    - suitesparse 4.5.*
+    - zlib 1.2.*
+    - bzip2 1.0.*
+    - portaudio 19
+    - libsndfile 1.0.27
+    - qscintilla2 2.9.*
+    # Following gets linked at the end.
+    - icu 58.*
+    - ncurses 5.9*
+    - xz 5.2.*
+    - libxml2 2.9.*
+    - libpng >=1.6.28,<1.7
+    - glib 2.51.*
+    - qt 4.8.*
+    - openssl 1.0.*
+  run:
+    - openblas 0.2.19|0.2.19.*
+    - blas 1.1 {{ variant }}
+    - pcre >=8.37,<9
+    - readline 6.2*
+    - arpack 3.4.*
+    - curl 7.*
+    - fftw 3.3.*
+    - fontconfig 2.12.*
+    - freetype 2.7|2.7.*
+    - glpk 4.61
+    - gnuplot 5.0.*
+    - hdf5 1.8.17|1.8.17.*
+    - suitesparse 4.5.*
+    - zlib 1.2.*
+    - bzip2 1.0.*
+    - portaudio 19
+    - libsndfile >=1.0.27,<1.1
+    - qscintilla2 2.9.*
+    # Following gets linked at the end.
+    - icu 58.*
+    - ncurses 5.9*
+    - xz 5.2.*
+    - libxml2 2.9.*
+    - libpng >=1.6.28,<1.7
+    - glib 2.51.*
+    - qt 4.8.*
+    - openssl 1.0.*
+    - libgfortran    # [osx]
```

Comment:
If it links to `libgomp` too we may need `libgcc`. I'll check the inspect linkages in a moment.